pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go

```diff
@@ -115,18 +154,33 @@ type BasicAutoScalerSpec struct {
 	// +optional
 	ScaleOutIntervalSeconds *int32 `json:"scaleOutIntervalSeconds,omitempty"`
 
+	// Deprecated
 	// +optional
 	Metrics []CustomMetric `json:"metrics,omitempty"`
 
 	// MetricsTimeDuration describes the Time duration to be queried in the Prometheus
+	// Deprecated
 	// +optional
 	MetricsTimeDuration *string `json:"metricsTimeDuration,omitempty"`
 	// ExternalEndpoint makes the auto-scaler controller able to query the external service
 	// to fetch the recommended replicas for TiKV/TiDB
 	// +optional
-	ExternalEndpoint *ExternalEndpoint `json:"externalEndpoint,omitempty"`
+	ExternalEndpoint *ExternalConfig `json:"external,omitempty"`
+}
+
+// +k8s:openapi-gen=true
+// ExternalConfig represents the external config.
+type ExternalConfig struct {
+	// ExternalEndpoint makes the auto-scaler controller able to query the
+	// external service to fetch the recommended replicas for TiKV/TiDB
+	// +optional
+	Endpoint *ExternalEndpoint `json:"endpoint"`
+	// maxReplicas is the upper limit for the number of replicas to which the autoscaler can scale out.
+	// It cannot be less than minReplicas.
```

Comment:
We can remove this as no minReplicas anymore.

Reply:
Done